### PR TITLE
Fix embeded_params method for all associations

### PR DESCRIPTION
**Pull request by raisin**
The embeded_params method was fixed to properly handle all associations
including the has_one, has_many, and belongs_to. In addition, the
class_name was required to fix two specs. This problem in the specs
did not previously surface because the has_one and belongs_to
associations were being ignored.

### CODE REVIEW

**Comment by raisin** [comment]
This pull request was rebased against the current master branch on June 20th, 2017. Afterwards, the specs were run to ensure they still pass, and the code was checked to see if the problem still exists without the changes made to lib/her/mode/parse.rb. The following three specs, which were added, still won't pass without the minor changes that were made:

spec/model/associations_spec:306
spec/model/associations_spec:312
spec/model/associations_spec:419

Would someone take a look at this pull request? Thanks

File: lib/her/model/parse.rb [hunk 1]
```diff
@@ -54,16 +54,20 @@ def to_params(attributes, changes={})
...
         def embeded_params(attributes)
-          associations[:has_many].select { |a| attributes.include?(a[:data_key])}.compact.inject({}) do |hash, association|
-            params = attributes[association[:data_key]].map(&:to_params)
-            next hash if params.empty?
-            if association[:class_name].constantize.include_root_in_json?
-              root = association[:class_name].constantize.root_element
-              hash[association[:data_key]] = params.map { |n| n[root] }
-            else
-              hash[association[:data_key]] = params
+          associations.values.flatten.select { |a| attributes.include?(a[:data_key]) }.compact.inject({}) do |hash, association|
```
Comment (zacharywelch) [inline lib/her/model/parse.rb]: Can we try iterating over `association_names` instead?

File: spec/model/associations_spec.rb [hunk 2]
```diff
@@ -403,6 +415,12 @@
       expect(params[:comments]).to be_kind_of(Array)
       expect(params[:comments].length).to eq(2)
     end
+
+    it "includes belongs_to relationship in params by default" do
```
Comment (zacharywelch) [inline spec/model/associations_spec.rb]: do we need a `has_one` spec also?

File: lib/her/model/parse.rb (hunk 1 above)
Comment (raisin) [inline lib/her/model/parse.rb]: Sure. I modified the code to iterate over association_names.

File: spec/model/associations_spec.rb (hunk 2 above)
Comment (raisin) [inline spec/model/associations_spec.rb]: I added a has_one spec.

**Comment by raisin** [comment]
@zacharywelch Thanks for looking at my PR! I'm looking forward to getting this merged. Also, I want to thank you for all the work you've done maintaining this gem. :)

**Comment by zacharywelch** [comment]
merged in a29c883f2ed875908825e06b4f714f79d0116e41. Made a few small adjustments to try and take advantage of the `associations` array. Thanks again for your help 😺

**Comment by raisin** [comment]
Thanks for looking at the PR, making modifications, and merging. Your
changes made the code more elegant.

**Pull request closed without merging**